### PR TITLE
US001 - View Job Roles

### PR DESCRIPTION
**Pull request by OdhranHague**
As a Kainos employee

I want to view job roles within Kainos
So that I can see all the job roles available in Kainos

Acceptance Criteria

Can view a list of job roles within Kainos

Tech notes:
Job Roles table
API endpoint GET /job-roles
UI page to display results from API

Added:
endpoint for /job-roles
roleController, roleService, roleDao and Role class matching attributes that are in DB

Screenshot 2023-12-06 at 10 32 49

Screenshot 2023-12-06 at 10 35 29

### CODE REVIEW

**Review by Joseph-Magee — APPROVED** [review]
looks good to me

**Comment by jkelso662** [comment]
General comment re screenshots, your pull request has all the code to view. Might be more useful to see swagger UI. Just a thought for next time!

**Comment by lullabelle** [comment]
Hi
Can you remove placeholder text files and any extensions from gitignore that dont need to be there (NetBeans, Gradle and Eclipse)

**Review by jkelso662 — APPROVED** [review]
Couple of things to watch out for (look at the two comments) but otherwise looking great! Can't wait to see how the tester gets on :)

**Comment by OdhranHague** [comment]
> Hi Can you remove placeholder text files and any extensions from gitignore that dont need to be there (NetBeans, Gradle and Eclipse)

done now :)

**Comment by OdhranHague** [comment]
finally got linter working. Used to a new one called checkstyle

File: src/main/java/org/kainos/ea/JobForgeWebServiceConfiguration.java [hunk 1]
```diff
@@ -3,18 +3,18 @@
...
 
 import javax.validation.Valid;
-import javax.validation.constraints.*;
+import javax.validation.constraints.NotNull;
 
 public class JobForgeWebServiceConfiguration extends Configuration {
     @Valid
-    @NotNull
-    private final SwaggerBundleConfiguration swagger = new SwaggerBundleConfiguration();
+    @NotNull // Test to see if it passes linter
```
Comment (jkelso662) [inline src/main/java/org/kainos/ea/JobForgeWebServiceConfiguration.java:12]: Worth probably dropping this.

**Review by jkelso662 — APPROVED** [review]
Looking great! Just watch out for that forgotten comment but otherwise LGTM!

**Review by Criofann — APPROVED** [review]
Comprehensive pull request and everything looks good to me

**Comment by jkelso662** [comment]
Was this user story signed off by the customer!?

**Pull request merged**